pubsub/google/cloud/pubsub/topic.py

```diff
@@ -252,7 +252,10 @@ def publish(self, message, client=None, **attrs):
         api = client.publisher_api
 
         self._timestamp_message(attrs)
-        message_b = base64.b64encode(message).decode('ascii')
+        if client._use_gax:
+            message_b = message.decode('ascii')
+        else:
+            message_b = base64.b64encode(message).decode('ascii')
```

Comment:
Ugh!  Rather than switching here, let's just push the "how-to-encode" logic down into the JSON and GAPIC helpers.